src/System.Private.ServiceModel/src/System/ServiceModel/Security/SecurityProtocolFactory.cs

```diff
@@ -686,6 +693,31 @@ public virtual void OnClose(TimeSpan timeout)
             }
         }
 
+        public virtual Task OnCloseAsync(TimeSpan timeout)
+        {
+            return OnCloseAsyncInternal(timeout);
+        }
+
+        private async Task OnCloseAsyncInternal(TimeSpan timeout)
+        {
+            TimeoutHelper timeoutHelper = new TimeoutHelper(timeout);
+            if (!_actAsInitiator)
+            {
+                foreach (SupportingTokenAuthenticatorSpecification spec in _channelSupportingTokenAuthenticatorSpecification)
+                {
+                    await SecurityUtils.CloseTokenAuthenticatorIfRequiredAsync(spec.TokenAuthenticator, timeoutHelper.RemainingTime());
+                }
+                foreach (string action in _scopedSupportingTokenAuthenticatorSpecification.Keys)
```

Comment:
(though I get that we want to keep the code similar to the sync path...) Feel free to not fix this 

---
In reply to: [78406802](https://github.com/dotnet/wcf/pull/1515#discussion_r78406802) [](ancestors = 78406802)